### ISSUE TITLE
App suddenly in background

### ISSUE THREAD

**Issue by jfeil** [allRisc/RefWatchPlus#15]
At first I wanna acknowledge this won't be a very good first issue, it's more like for documentation purposes if it happens again, as I don't know how to reproduce :)

During one of my last matches I had to switch position (first linesman to referee) and somehow suddenly the app itself was in the background? I don't know if I pressed any random inputs, during switching the flag receiver. During the break I paused for correct overtime and it was somehow around 5min as it broke.
I would see the native activity overview (as like I'm currently running), but couldn't stop it either as the app still reacted on my inputs... (But I did no longer see overtime and current time and so on, just the native activity screen..)

I'm using a forerunner 955 with the latest version installed (12.xx)

**Comment by allRisc** [comment]
When you say the native activity overview do you mean the one in the app or the one built into the forerunner?

**Comment by jfeil** [comment]
It was the one built into the forerunner, the one if I start a running activity manually

**Comment by allRisc** [comment]
Very strange. I am not aware of any path through the event handling logic which would enable the activity to still run, but exit the app. Let me dive into this a little bit.

**Comment by jfeil** [comment]
As I said before: The app was not really exited as it still reacted on button presses... It was just somehow in the background...

It is a also a possibility that this is not an issue with the app itself but with the OS...

**Comment by brentcwilson** [comment]
I'm not certain, but what you describe sounds somewhat like something I experienced during a match. Perhaps not coincidentally, I too am running the app on a 955. Unfortunately I can't provide much detail that would be useful for debugging, however I am reffing a _lot_ of matches this fall, so perhaps it will happen again so that I can document the experience.

**Comment by brentcwilson** [comment]
Following up here:

I have officiated 40-ish matches since my previous comment. I have not experienced the same hiccup again.

**Comment by jfeil** [comment]
Yeah, same for me.. did not happen any time after this accident... I just don't know what the reasons were 🥲

**Comment by jfeil** [comment]
It somehow happened again last Saturday! During the halftime break the watch was suddenly at the watchface and when I pressed the start/stop key, it switched to the activity screen and paused it. But I couldn't get back into the app... It somehow might again happened by pressing accidentally some buttons (I was wearing many shirts as it was close to 0°C).

**Comment by allRisc** [comment]
Is there an app crash log? Like the one described: https://developer.garmin.com/connect-iq/core-topics/debugging/

**Comment by jfeil** [comment]
I checked and the only ones were from 2nd of October, when there was this faulty build :/

**Comment by allRisc** [comment]
Bummer, that's what I thought might happen since it's not a crash...

Looking through the API docs the app may be getting suspended and this suspension isn't being handled properly.

I will look into it later this week
